### DISCUSSION TITLE
How to use AG grid API methods in NiceGUI table

### DISCUSSION THREAD

**Discussion by JS-Aibel** [zauberzeug/nicegui#170] [Q&A] [answered]
How do we access the selected rows (checkboxes) from Nice GUI?
The table below is made with Nice GUI.

![image](https://user-images.githubusercontent.com/100705556/203022934-26e1dca3-af6c-44d2-9565-51c7a469334a.png)

**Start code**

```python
from nicegui import ui

table = ui.table({
    'columnDefs': [
        {
            'headerName': 'Athlete', 'field': 'athlete',
            'headerCheckboxSelection': True,
            'checkboxSelection': True,
            'showDisabledCheckboxes': True,
        },
        {'headerName': 'Sport', 'field': 'sport'},
        {'headerName': 'Year', 'field': 'year', 'maxWidth': 120},
    ],
    'rowData': [
        {'athlete': 'Alice', 'sport': 'soccer', 'year': 1980},
        {'athlete': 'Bob', 'sport': 'golf', 'year': 1980},
        {'athlete': 'Bill', 'sport': 'tennis', 'year': 2012},
    ],
    'defaultColDef': {
        'flex': 1,
        'minWidth': 100,
    },
    'rowSelection': 'multiple',
    'suppressRowClickSelection': True,
})

ui.run()
```

**Comment by falkoschindler** [comment]
Currently there is only limited support for AG Grid API methods, since JustPy's `run_api` doesn't seem to return anything. [There](https://justpy.io/grids_tutorial/grid_api/) is, however, an example showing how to subscribe to row selection events and storing the selected rows in Python. In NiceGUI the event subscription might look as follows:

```python
table.view.on('rowSelected', lambda _, msg: print(msg['rowIndex'], msg['selected'], flush=True))
```

**Reply by me21** [reply]
Try the following code, it worked for me:

```python
async def print_rowdata():
    rowdata = await table.page.run_javascript(f"""cached_grid_def['g' + {table.view.id}].rowData""", timeout=60)
    print(rowdata)
```

It returns the whole table, though. Maybe you could tailor the example by referencing another data member, calling API, etc.
It is basically copied from JustPy, only we explicitly store the result (and call NiceGUI's implementation of `run_javascript`, not JustPy's).

**Comment by JS-Aibel** [comment, marked as answer]
Thank you.
We did this: 
```python
table.view.on('cellValueChanged', checkBoxChanged)

def checkBoxChanged(sender, msg):
  if msg.colId == 'enabled':
    table.options.rowData[msg.rowIndex]['enabled'] = msg.newValue
```

Then we printed all the rows in rowData that was set to true.
```python
        if row['enabled']==True:
          rowlist.append(row)
```
